### PR TITLE
Clean up the rest of the Map grossness left over from DDC workarounds

### PR DESCRIPTION
**Pull request by greglittlefield-wf**
https://github.com/Workiva/over_react/pull/155 did much of the work around cleaning up our internal Map classes, and there were just a few more things left that could be removed, which is this PR.

Testing (in Dart 2):
- Verify that the example `pub build`s with no compile-time errors
- Verify that the example has no compile-time or runtime errors in the DDC

### CODE REVIEW

**Comment by aviary-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://dev.workiva.net/docs/information-security/aviary#configuring-aviary) coverage of security relevant code
---
_Questions or Comments? Reach out on HipChat: InfoSec Forum._

File: lib/src/component_declaration/component_base.dart [hunk 1]
```diff
@@ -459,13 +459,7 @@ abstract class UiStatefulComponent<TProps extends UiProps, TState extends UiStat
 ///
 /// > Note: Implements [MapViewMixin] instead of extending it so that the abstract state declarations
 /// don't need a constructor. The generated implementations can mix that functionality in.
-abstract class UiState extends _OverReactMapView implements StateMapViewMixin {
-  // Manually implement members from `StateMapViewMixin`,
-  // since mixing that class in doesn't play well with the DDC.
-  // TODO find out root cause and reduced test case.
-  @override Map get _map => this.state;
-  @override String toString() => '$runtimeType: ${prettyPrintMap(_map)}';
```
Comment (aaronlademann-wf) [inline lib/src/component_declaration/component_base.dart]: @greglittlefield-wf I'm assuming that `toString` still returns `'$runtimeType: ${prettyPrintMap(_map)}'` in the super class?

**Comment by aaronlademann-wf** [comment]
+1

File: lib/src/component_declaration/component_base.dart (hunk 1 above)
Comment (greglittlefield-wf) [reply lib/src/component_declaration/component_base.dart]: Yup, that should get mixed in with `PropsMapViewMixin`

**Pull request merged**